src/runtime_src/core/common/device.cpp

```diff
@@ -145,64 +147,121 @@ xrt::xclbin
 device::
 get_xclbin(const uuid& xclbin_id) const
 {
-  if (xclbin_id && xclbin_id != m_xclbin.get_uuid())
-    throw error(EINVAL, "xclbin id mismatch");
+  // Allow access to xclbin in process of loading via device::load_xclbin
+  if (xclbin_id && xclbin_id == m_xclbin.get_uuid())
```

Comment:
Tricky, but no.  m_xclbins may not yet have been populated.  The xclbin is in the process of being loaded.   I don't particular fancy this code, and loading is convoluted because it has to support user applications calling xclLoadXclbin directly and native APIs.